ldj-spring-renderer-example/src/main/java/com/infosupport/ldoc/springexample/ampq/SpringAmpqRenderer.java

```diff
@@ -0,0 +1,234 @@
+      String className = typeDesc.path("FullName").textValue();
+
+      //check if class is listener
+      JsonNode rabbitListenerAnnotation = containsAttribute(typeDesc, RABBIT_LISTENER_ANNOTATION);
+      if (rabbitListenerAnnotation != null) { // I.e., this class is a RabbitListener
+        // Note that the queue is the same for all methods within a RabbitListener
+        JsonNode queueArgument = getNodeWithKeyAndValue(rabbitListenerAnnotation.path("Arguments"),
+            "Name", "queues");
+        assert queueArgument
+            != null; // TODO This should not happen, how nicely do we want to handle this?
+        String queue = queueArgument.path("Value").asText();
+
+        //Check for listen methods
+        for (JsonNode method : typeDesc.path("Methods")) {
+          if (isQueueReadMethod(method)) {
+            // Note the type is stripped to support messages across different applications
+            String messageType = stripName(
+                method.path("Parameters").path(0).path("Type").textValue());
+            // Create and add ReadInteraction
+            QueueInteraction newRead = new QueueInteraction(className, QueueInteractionKind.READ,
+                queue, messageType, new ArrayList<>(0));
+            // Add all posts that happen within this read
+            newRead.reactionToRead().addAll(findQueuePosts(method, className));
+
+            interactions.add(newRead);
+          } else {
+            // Check for non-listening methods whether posts are made
+            interactions.addAll(findQueuePosts(typeDesc, className));
+          }
+        }
+      } else {
+        // Check for non-listener classes
+        interactions.addAll(findQueuePosts(typeDesc, className));
+      }
+
+    }
+    return interactions;
+  }
+
+
+  /**
+   * Recursive method to create all PlantUML code for all interactions.
+   *
+   * @param out                             writer where PlantUML is written to
+   * @param allInteractions                 all interactions associated with the source
+   *                                        applications
+   * @param relevantPostInteractionsToReply list of Post Queue interactions that should be rendered
+   *                                        for this cycle
+   */
+  static void renderInteractions(PrintWriter out, List<QueueInteraction> allInteractions,
+      List<QueueInteraction> relevantPostInteractionsToReply) {
+    // Loop over all top-level post interactions
+    for (QueueInteraction postInteraction : relevantPostInteractionsToReply) {
+      // Loop over all read reactions that get triggered after the postInteraction
+      for (QueueInteraction readInteraction : postInteraction.filterReadsReactingToThis(
+          allInteractions)) {
+        PlantUmlBuilder.renderInteraction(out, postInteraction.actor(), readInteraction.actor(),
+            readInteraction.messageType());
+
+        out.println("activate %s".formatted(readInteraction.actor()));
```

Comment:
Agreed. But I waited with this because this code change is not final (i.e. wait for the decision of how we want the sample renderers to be structured)